dollar_ref/__init__.py

```diff
@@ -191,7 +191,8 @@ def read_file(path: str) -> dict:
         raw = file.read()
 
         try:
-            if raw.startswith('---'):
+            if raw.startswith('---') or any(
+                    [path.lower().endswith(s) for s in ['.yaml', '.yml']]):
```

Comment:
*minor:* would be great to have some brief comment on this, just like you described in the PR description.